core/server/i18n.js

```diff
@@ -94,23 +121,88 @@ I18n = {
+        // full internationalization of core and theme.
+
+        var blosTheme, hasBuiltInLocaleData, IntlPolyfill, backendTranslations;
+
+        if (blos === undefined) {
+            // If not in memory, load translations for core and theme.
+            i18nCore = true;
+            i18nTheme = true;
+        }
+
+        // Switch to enable/disable translations of back-end messages from core .js files.
+        // - true: enabled; optional files like content/locales/es.json, etc. are used when available.
+        // - false: disabled; only the default English file core/server/translations/en.json is used.
+        // Before enabling back-end message translations, see previous tasks on issue #6526
+        // (error codes or identifiers, error message translation at the point of display...)
+        backendTranslations = false;
+
+        currentLocale = settingsCache.get('default_locale') || 'en';
+        activeTheme = settingsCache.get('active_theme') || '';
+
+        // Reading files for current locale and active theme and keeping their content in memory
+        // (during Ghost's initialization, when English is the language in settings, the second call
+        // to i18n.init() doesn't read again the default English file for core if already in memory).
+        if (i18nCore) {
+            // Reading translation file for core .js files.
+            if (currentLocale === 'en' || !backendTranslations) {
+                blos = fs.readFileSync(path.join(__dirname, 'translations', 'en.json'));
+            } else {
+                // Preventing missing file.
+                try {
+                    blos = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'locales', currentLocale + '.json'));
+                } catch (err) {
+                    if (err.code === 'ENOENT') {
+                        logging.warn('File content/locales/' + currentLocale + '.json not found! Falling back to default core/server/translations/en.json.');
+                        blos = fs.readFileSync(path.join(__dirname, 'translations', 'en.json'));
+                    } else {
+                        throw err;
+                    }
+                }
+            }
+            // if translation file is not valid, you will see an error
+            try {
+                blos = JSON.parse(blos);
+            } catch (err) {
+                blos = undefined;
+                throw err;
+            }
+        }
+        if (i18nTheme && activeTheme) {
+            // Reading translation file for theme .hbs templates.
+            // Compatibility with both old themes and i18n-capable themes.
+            // Falling back any wrong locale to default English for this session.
+            try {
+                blosTheme = fs.readFileSync(path.join(__dirname, '..', '..', 'content', 'themes', activeTheme, 'assets', 'locales', currentLocale + '.json'));
```

Comment:
Do you have any reference material to put the themes into `assets`?

Otherwise, a theme `locales` folder should be located in `content/themes/{my-theme}/locales`, [see](https://github.com/TryGhost/Ghost/pull/8437#issuecomment-330309612).

Reply:
Done this change with today's commit:
[i18n] Theme's _assets/locales_ folder moved to _locales_

Also, updated the [help file](https://github.com/juan-g/Ghost/blob/i18n-translatable-frontend/content/locales/TRANSLATIONS.md).